client/client.go

```diff
@@ -0,0 +1,425 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const (
+	baseURL            = "https://api.app.firebolt.io"
+	loginURL           = baseURL + "/auth/v1/login"
+	refreshTokenURL    = baseURL + "/auth/v1/refresh"
+	accountIDByNameURL = baseURL + "/iam/v2/accounts:getIdByName?account_name=%s"
+	engineIDByNameURL  = baseURL + "/core/v1/accounts/%s/engines:getIdByName?engine_name=%s"
+	engineURLByNameURL = baseURL + "/core/v1/accounts/%s/engines?filter.name_contains=%s"
+	engineByIDURL      = baseURL + "/core/v1/accounts/%s/engines/%s"
+	startEngineURL     = baseURL + "/core/v1/accounts/%s/engines/%s:start"
+
+	// retryMax is the maximum number of retries.
+	retryMax = 3
+	// engineStatusCheckTimeout is a timeout for checking engine status.
+	engineStatusCheckTimeout = time.Second * 5
+)
+
+// Client client for calls to firebolt.
+type Client struct {
```

Comment:
i.e. there's no Go client we can use (open source, official and/or community driven)?

Reply:
We don't have any Go client in firebolt official documentation . But I have found in github this
https://github.com/firebolt-db/firebolt-go-sdk

Look likes it is fresh and in active development . Do we want try it?

Reply:
TBH, I'm a little bit skeptical because it's only 2 months old, and because of the note about unimplemented methods. So if we try it out, I think we should do that separately.